### PR TITLE
feat: allow selecting application on detail page

### LINKED ISSUE #7186: Quickly switch between applications, bypassing main page
# Summary

Would be nice to be able to search and jump to applications from application pages, not just from the main page.

# Motivation

We have a lot of apps, and from some parts of the world loading /applications takes upwards of 15 seconds (some days much more, depending on internet quality), so jumping back to the main applications page every time I need to switch to another app annoying to say the least (a real productivity killer). 

See https://github.com/argoproj/argo-cd/discussions/7098

# Proposal

Would be nice to have a 'search' box on every application-details page, like on the main page. Or some other "quick switcher".

### PR DESCRIPTION
**Pull request by alexmt**
Signed-off-by: Alexander Matyushentsev <AMatyushentsev@gmail.com>

Depends on https://github.com/argoproj/argo-ui/pull/175
Closes #7186

https://user-images.githubusercontent.com/426437/149423433-328fa6cd-870d-449d-a518-47ba937c866b.mov

PR allows changing the navigating to the application by name from the Application Details page.

### CODE REVIEW

**Comment by alexmt** [comment]
@ciiay, @rbreeze  can you take a look please?

File: ui/package.json [hunk 1]
```diff
@@ -28,7 +28,7 @@
     "@types/react-router-dom": "^4.2.3",
     "@types/superagent": "^3.5.7",
     "ansi-to-react": "^6.1.6",
-    "argo-ui": "git+https://github.com/argoproj/argo-ui.git",
+    "argo-ui": "git+https://github.com/alexmt/argo-ui.git#customizable-page-breadcrumbs",
```
Comment (keithchong) [inline ui/package.json:31]: Hi Alex, I thought only the lock file was needed to be changed to pick up the changes in argo-ui

File: ui/src/app/applications/components/application-details/application-details.tsx [hunk 2]
```diff
@@ -198,9 +199,13 @@ export class ApplicationDetails extends React.Component<RouteComponentProps<{nam
...
                                 <div className='application-details'>
                                     <Page
-                                        title='Application Details'
+                                        title={`${this.props.match.params.name} / Application Details`}
+                                        useTitleOnly={true}
                                         toolbar={{
-                                            breadcrumbs: [{title: 'Applications', path: '/applications'}, {title: this.props.match.params.name}],
+                                            breadcrumbs: [
+                                                {title: 'Applications', path: '/applications'},
+                                                {title: <ApplicationsDetailsAppDropdown appName={this.props.match.params.name} />}
```
Comment (keithchong) [inline ui/src/app/applications/components/application-details/application-details.tsx:207]: The title is used as the key, which is supposed to be a string, for each breadcrumb path in top-bar.

File: ui/package.json (hunk 1 above)
Comment (alexmt) [reply ui/package.json]: I had to temporary change package.json to use my fork with un-merged changes. Reverted changes.

File: ui/src/app/applications/components/application-details/application-details.tsx (hunk 2 above)
Comment (alexmt) [reply ui/src/app/applications/components/application-details/application-details.tsx:223]: Thank you for catching it! These is a bug in `argo-ui`. Created PR that fixes it and allows using `React.Node`: https://github.com/argoproj/argo-ui/pull/176

**Review by ciiay — APPROVED** [review]
Looks great to me! Love this solution 👍

File: ui/src/app/applications/components/application-details/application-details.tsx (hunk 2 above)
Comment (keithchong) [reply ui/src/app/applications/components/application-details/application-details.tsx:223]: Hi @alexmt, that change in 176 looks ok to me.

**Review by rbreeze — APPROVED** [review]
LGTM! Awesome!

**Pull request merged**